### ISSUE TITLE
Joi.Forbidden() should prevent property from being output in typescript 

### ISSUE THREAD

**Issue by VagabondOfHell** [mrjono1/joi-to-typescript#112]
When Forbidden and/or Strip is used, the resulting typescript model should not have the entry defined in its model.

```
export const examplePOSTBodySchema = baseSchema
  .fork(['id'], (x) => x.forbidden().strip(true))
  .meta({className: 'examplePOSTBody'});
```
Here I am forking a base schema and trying to remove the id parameter. Unfortunately Forbidden is not supported so id is still output in the resulting model.

**Comment by mrjono1** [comment]
HI
`.forbidden()` hasn't been implemented yet, I'm open for a PR to support this feature

**Comment by mrjono1** [comment]
I haven't used Joi.forbidden() you also have used Joi.strip() here are you able to provide an issue using the template including the actual and expected output?

**Comment by VagabondOfHell** [comment]
I used strip and forbidden just to see if any happened to be supported in the output. I was just looking to define one base model that would be slightly modified based on its intended usage.

``` 
export const offerSchema = Joi.object({
  id: Joi.string().guid(),
  advertiserId: Joi.string(),
  advertiser: Joi.string(),
  advertiserType: Joi.string().valid('retailer', 'property'),
  priority: Joi.string().valid('standard', 'fallback'),
  data: Joi.array().items(offerDataSchema.required()).min(1),
  schedule: offerScheduleSchema,
  displaysAt: Joi.array().items(offerDisplayTargetSchema.required()).min(1)
});


export const offerPOSTBodySchema = offerSchema
  .fork(['id'], (x) => x.forbidden().strip(true))
  .fork(['advertiserId', 'advertiser', 'advertiserType', 'priority', 'data', 'schedule', 'displaysAt'], (x) => x.required())
  .meta({className: 'OfferPOSTBody'});
```

```
export interface OfferPOSTBody {
  id?: string;  <--- Ideally this wouldn't be here at all in this case using either strip or forbidden
  advertiserId: string;
  advertiser: string;
  advertiserType: string;
  priority: string;
  data: OfferDataDTO[];
  schedule: OfferScheduleDTO;
  displaysAt: OfferTargetDTO[];
}
```

**Comment by shennan** [comment]
I have a scenario where I use `Joi.forbidden()` to indicate that the value should be `undefined`.

So my expected behaviour would be:

```
const RequestBodySchema = Joi.forbidden()

RequestBodySchema.validate(undefined) // Passes
RequestBodySchema.validate({}) // Error
RequestBodySchema.validate({ foo: 'bar' }) // Error

```

With this in mind, I think `Joi.forbidden()` should actually compile to:

```
type RequestBodySchema = never
```

Any thoughts on this? Currently it just compiles to `any`, which I think is wrong.

**Comment by mrjono1** [comment]
`never` I like that

**Comment by shennan** [comment]
Would you be happy with a PR to that effect?

**Comment by mrjono1** [comment]
Yes please

**Comment by mrjono1** [comment]
I have created a PR for this https://github.com/mrjono1/joi-to-typescript/pull/217

**Comment by mrjono1** [comment]
Fixed in 4.0.0

**Comment by shennan** [comment]
This doesn't work in latest version (not tested 4.0.0).

This:
```
const SomeForbiddenSchema = Joi.forbidden()
```

Produces this:
```
export type undefined = any;
```
